pkg/tsdb/sql_engine.go

```diff
@@ -163,6 +172,33 @@ func (e *sqlQueryEndpoint) Query(ctx context.Context, dsInfo *models.DataSource,
 	return result, nil
 }
 
+// global macros/substitutions for all sql datasources
+func Interpolate(query *Query, timeRange *TimeRange, sql string) (string, error) {
+	rExp, _ := regexp.Compile(`\$__(interval|interval_ms)\b`)
+	var err error
+
+	sql = ReplaceAllStringSubmatchFunc(rExp, sql, func(groups []string) string {
+		switch groups[1] {
+		case "interval":
+			interval, err := GetIntervalFrom(query.DataSource, query.Model, time.Second*60)
```

Comment:
This only returns the minInterval. Same applies for interval_ms.

You'll need to use the [IntervalCalculator](https://github.com/grafana/grafana/blob/master/pkg/tsdb/interval.go#L52) to get the actual interval. See [example usage for influxdb](https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/query.go#L32-L38). 

Optional, but very appreciated, it would be very nice if we could add a Milliseconds() method to [Interval](https://github.com/grafana/grafana/blob/master/pkg/tsdb/interval.go#L19). Then we can use that method here and for [elasticsearch](https://github.com/grafana/grafana/blob/master/pkg/tsdb/elasticsearch/client/client.go#L141) and [influxdb](https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/query.go#L37).